### PR TITLE
Load GAP Development manual if it is present

### LINKED ISSUE #1161: Loading developers manual
Question: would it be useful if doc/dev manual would be also built by `make manuals` if present, and would be also loaded, if it's present (so `?` and `??` would also search in the doc/dev manual)? 

`doc/dev` is not a part of the GAP distribution, but perhaps developers would appreciate being able to search in it (e.g. there is a section on GASMAN there) and maybe that would help to keep it more up-to-date and fill in missing holes?

### PR DESCRIPTION
**Pull request by alex-konovalov**
This is an updated version of #1194. It closes #1161.

### CODE REVIEW

File: lib/helpbase.gi [hunk 1]
```diff
@@ -493,12 +493,16 @@ InstallValue(HELP_KNOWN_BOOKS, [[],[]]);
 
 
 InstallGlobalFunction(HELP_ADD_BOOK, function( short, long, dir )
-  local sortfun, str, hnb, pos;
+  local main, sortfun, str, hnb, pos;
+  main := ["tutorial", "reference", "changes"];
+  if ForAny( GAPInfo.RootPaths,
+             rp -> IsDirectoryPath( Concatenation( rp, "/doc/dev" ) ) ) then
+    Add( main, "development" );
+  fi;
```
Comment (fingolfin) [inline lib/helpbase.gi]: Why is this needed? can't you just add `"development"` to the list inside `sortfun` and be done with it?

Comment (alex-konovalov) [reply lib/helpbase.gi]: This is needed because we do not include this manual in the GAP distribution. Done outside `sortfun` accordingly to your comment in https://github.com/gap-system/gap/pull/1194

File: lib/helpbase.gi [hunk 2]
```diff
@@ -493,12 +493,16 @@ InstallValue(HELP_KNOWN_BOOKS, [[],[]]);
...
   # we sort books with main books first and packages alphabetically,
   # (looks a bit lengthy)
   sortfun := function(a, b)
-    local main, pa, pb;
-    main := ["tutorial", "reference", "changes"];
```
[the remark this answers is not in the document]
Comment (alex-konovalov) [reply lib/helpbase.gi]: Right, we check whether the book exist when we call `HELP_ADD_BOOK`. But then perhaps we need to add `"HPC-GAP"` here as well. I will revise this at the 2nd half of the week.

[the remark this answers is not in the document]
Comment (alex-konovalov) [reply lib/helpbase.gi]: @fingolfin @ChrisJefferson: The fact that there is no need to check inside `sortfun` was not picked up while reviewing the previous version of this PR, so it escaped my attention.  Now I've updated this PR and it's ready to be merged.

File: lib/helpbase.gi [hunk 3]
```diff
@@ -498,7 +498,7 @@ InstallGlobalFunction(HELP_ADD_BOOK, function( short, long, dir )
   # (looks a bit lengthy)
   sortfun := function(a, b)
     local main, pa, pb;
-    main := ["tutorial", "reference", "changes"];
+    main := ["tutorial", "reference", "changes", "HPC-GAP", "development" ];
```
Comment (fingolfin) [inline lib/helpbase.gi]: Was this tested? Shouldn't all entries be lower case, to match the output of `SIMPLE_STRING`?

Comment (alex-konovalov) [reply lib/helpbase.gi]: yes - for example, in HPC-GAP we have for `??atomic`:

```
[1] Reference: atomic irrationalities
[2] HPC-GAP: The atomic statement.
[3] HPC-GAP: Atomic functions
[4] HPC-GAP: Atomic objects
[5] HPC-GAP: Atomic lists
[6] HPC-GAP: Atomic records and component objects
[7] HPC-GAP: AtomicIncorporateObj
[8] HPC-GAP: AtomicList
[9] HPC-GAP: AtomicList for a count and an object
[10] HPC-GAP: FixedAtomicList
[11] HPC-GAP: FixedAtomicList for a count and an object
[12] HPC-GAP: MakeFixedAtomicList
[13] HPC-GAP: FromAtomicList
[14] HPC-GAP: ATOMIC_ADDITION
[15] HPC-GAP: AtomicRecord
[16] HPC-GAP: AtomicRecord for a record
[17] HPC-GAP: FromAtomicRecord
[18] HPC-GAP: MakeWriteOnceAtomic
[19] HPC-GAP: MakeStrictWriteOnceAtomic
[20] HPC-GAP: MakeReadWriteAtomic

```
and for `??GASMAN`:

```
[1] Reference: GASMAN
[2] Development: GASMAN
[3] Reference: The GASMAN Interface for Weak Pointer Objects
[4] Reference: GasmanStatistics
[5] Reference: GasmanMessageStatus
[6] Reference: SetGasmanMessageStatus
[7] Reference: GasmanLimits
[8] Development: GASMAN Interface
[9] Development: One More Bit of GASMAN interface
```

OTOH, other books are added with commands like `HELP_ADD_BOOK("Tutorial", "GAP 4 Tutorial", "doc/tut");` and then their name is used in lowercase in sortfun, so I will update the spelling for consistency.

File: lib/helpbase.gi [hunk 4]
```diff
@@ -498,7 +498,7 @@ InstallGlobalFunction(HELP_ADD_BOOK, function( short, long, dir )
   # (looks a bit lengthy)
   sortfun := function(a, b)
     local main, pa, pb;
-    main := ["tutorial", "reference", "changes"];
```
Comment (fingolfin) [inline lib/helpbase.gi]: Unfortunately, this PR did not apply these changes to the HPC-GAP version of the code :/. I missed that during review.

Comment (alex-konovalov) [reply lib/helpbase.gi]: @fingolfin you're right - I will reopen #1161 to remind me of this.

Comment (alex-konovalov) [reply lib/helpbase.gi]: For the record, this is now done by @fingolfin in #2054
